### DISCUSSION TITLE
Support RRULES

### DISCUSSION THREAD

**Discussion by benat974** [go-co-op/gocron#199] [Ideas]
Hey, I am a french engineer and I use your go-cron package for few IOT project. Until this day, I manipulated weekly schedule and it's was easy to convert it and create cron job with your package. Now I want to use [Rrules](https://datatracker.ietf.org/doc/html/rfc5545#section-3.3.10) to have more flexible possibilities to program an event. 

I  found some go package to convert rrules  to list of datetime (https://github.com/teambition/rrule-go). But there are not go package to program cron task from rrules. It will be a great feature for your package. I see in your issues queue [feature](https://github.com/go-co-op/gocron/issues/128) which can be manage via RRULE. I would like to have your feedback about this idea. 

I haven't access to your slack to discuss with us about it. I hope have news soon
Ben

**Comment by JohnRoesler** [comment]
Looks very interesting! I could see it being implemented in a similar way to crontabs where we use a service to tell us when the next scheduled time is. In this case, if it was designed such the job could keep a record of a rrule type scheduler which provided a method `Next()` it could be called by our scheduler. Just a thought, but actual implementation would need to be explored.

I think opening a feature request is a reasonable. I will say that I don't have the bandwidth to pick up something like this at the moment, but if you're interested in taking it on and proposing an implementation design I'd be happy to weigh in so you could then start implementing the code.

**Comment by benat974** [comment]
Finally I continue to use the cron format but I manage  metadata of an event (start and end datetime, excluded dates) in the callback to be able to have the flexibility of RRules format. One possible feature for you library is to be able to configure cron task from list of datetimes. It allows user to use another format of cron without support this format in your side.

Ben


Envoyé à partir d’Outlook<http://aka.ms/weboutlook>

________________________________
De : John Roesler ***@***.***>
Envoyé : lundi 21 juin 2021 18:25
À : go-co-op/gocron ***@***.***>
Cc : bnativel97400 ***@***.***>; Author ***@***.***>
Objet : Re: [go-co-op/gocron] Support RRULES (#199)


Looks very interesting! I could see it being implemented in a similar way to crontabs where we use a service to tell us when the next scheduled time is. In this case, if it was designed such the job could keep a record of a rrule type scheduler which provided a method Next() it could be called by our scheduler. Just a thought, but actual implementation would need to be explored.

I think opening a feature request is a reasonable. I will say that I don't have the bandwidth to pick up something like this at the moment, but if you're interested in taking it on and proposing an implementation design I'd be happy to weigh in so you could then start implementing the code.
